src/main/java/jackson/Jackson.java

```diff
@@ -0,0 +1,225 @@
+package jackson;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+
+import jackson.enums.Actions;
+import jackson.enums.Commands;
+import jackson.exceptions.OutOfListException;
+import jackson.exceptions.SyntaxException;
+import jackson.exceptions.UnsupportedException;
+import jackson.tasks.Deadline;
+import jackson.tasks.Event;
+import jackson.tasks.Task;
+import jackson.tasks.Todo;
+import jackson.utils.Parser;
+import jackson.utils.Response;
+import jackson.utils.Storage;
+import jackson.utils.TaskList;
+import jackson.utils.Ui;
+
+/**
+ * Main class for the chatbot.
+ */
+public class Jackson {
+
+    /* Expected number of tasks to store */
+    private static final int EXPECTED_SIZE = 100;
+
+    /* Path to save list data */
+    private static final String PATH = "src/main/resources/texts/data.txt";
+
+    /* Stores secret text for greedy loading */
+    private static String secret = "";
+
+    /* Stores previous command type for css style changing */
+    private Commands.CommandType commandType;
+
+    /* Instance variables for main loop */
+    private TaskList taskList;
+    private Ui ui;
+    private Storage storage;
+
+    /**
+     * Constructs Jackson instance.
+     */
+    public Jackson() {
+        this.taskList = new TaskList(EXPECTED_SIZE);
+        this.ui = new Ui();
+        this.storage = new Storage(PATH);
+        this.commandType = Commands.CommandType.INTRO;
+    }
+
+    /**
+     * Reads secret text from secret file and prints it.
+     * If secret file not found, handles exception and prints error message.
+     * @return String response.
+     */
+    public String readSecret() {
+        // get string builder to read line by line
+        StringBuilder temp = new StringBuilder();
```

Comment:
Maybe could rename this variable a bit to show what temp String we're using it for?